cmd/virt-launcher/virt-launcher.go

```diff
@@ -610,16 +616,34 @@ func RemoveContents(dir string) error {
 
 func terminateIstioProxy() {
 	if istioProxyPresent() {
-		resp, err := http.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", infraconfigurators.EnvoyMergedPrometheusTelemetryPort), "", nil)
-		if err != nil || resp.StatusCode != http.StatusOK {
-			log.Log.Error("Failed to request Istio proxy termination")
+		isRetriable := func(err error) bool {
+			if net.IsConnectionReset(err) || net.IsConnectionRefused(err) || k8serrors.IsServiceUnavailable(err) {
+				return true
+			}
+			return false
+		}
+		err := retry.OnError(retry.DefaultBackoff, isRetriable, func() error {
+			resp, err := httpClient.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", infraconfigurators.EnvoyMergedPrometheusTelemetryPort), "", nil)
+			if err != nil {
+				log.Log.Reason(err).Error("failed to request istio-proxy termination, retrying...")
+				return err
+			}
+			if resp.StatusCode != http.StatusOK {
+				log.Log.Errorf("failed to request istio-proxy termination, status code received: %d, retrying...", resp.StatusCode)
```

Comment:
@rhrazdil I believe what @maiqueb refers to is that on non-200 response, there's a log message saying "retrying..." but a `nil` err is returned hence no retry will occur. This can be resolved by either adjusting the log message, or adjusting the return value to a non-nil err. I think the two options are viable (I don't see how a 404 becomes a 200, but on the other hand no harm in retrying quickly).

Reply:
Ahh, just saw you changed the log message. Nevermind :-)